lib/src/main/java/uk/gov/di/data/lep/library/dto/DeathNotification/NamePart.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.di.data.lep.library.dto.DeathNotification;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+public record NamePart(
+    NamePartType type,
+    String value,
+    LocalDate validFrom,
+    LocalDate validUntil
+) {
+
+    public NamePart(NamePartType type, String value) {
+        this(type, value, null, null);
+    }
+
+    public enum NamePartType {
```

Comment:
I think I'd spin this out into it's own file and probably simplify it (e.g. remove the generated CONSTANTS, the `fromValue` etc)